lib/src/collect_lcov.dart

```diff
@@ -104,26 +105,23 @@ class LcovCollector {
   /// Generates and returns a coverage json file
   Future<CoverageResult<List<File>>> _getCoverageJson(
       String testFile, Directory coverageDir) async {
+    int observatoryPort = 8181; // FIXME: Should read out port instead?
     var args = [
-      "--coverage_dir=${coverageDir.path}",
-      "--package-root=${packageRoot}",
+      "--packages=${packagesPath}",
+      "--pause-isolates-on-exit",
+      "--enable-vm-service=${observatoryPort}",
       testFile
     ];
-    var result = await processSystem.runProcess(Platform.executable, args);
-    if (result.exitCode < 0) {
-      stderr.writeln('stdout:');
-      stderr.writeln(result.stdout);
-      stderr.writeln('stderr:');
-      stderr.writeln(result.stderr);
-      throw new ProcessException(
-          Platform.executable,
-          args,
-          'There was a critical error. Exit code: ${result.exitCode}',
-          result.exitCode);
-    }
+    Future<ProcessResult> testFuture =
+        processSystem.runProcess(Platform.executable, args);
+    final Map data = await collect('localhost', observatoryPort, true, false);
+
+    new File(p.join(coverageDir.path, 'coverage.json'))
+        .writeAsString(JSON.encode(data));
+
     var reportFiles =
         await coverageDir.list(recursive: false, followLinks: false).toList();
-    return new CoverageResult<List<File>>(reportFiles, result);
+    return new CoverageResult<List<File>>(reportFiles, await testFuture);
```

Comment:
Presumably when there is a syntax error we still need something to connect and resume the isolates?  Donno.